logic/suggestions/handToss.go

```diff
@@ -0,0 +1,72 @@
+package suggestions
+
+import (
+	"errors"
+
+	"github.com/joshprzybyszewski/cribbage/logic/scorer"
+	"github.com/joshprzybyszewski/cribbage/model"
+)
+
+func GetAllTosses(
+	hand []model.Card,
+) ([]model.TossSummary, error) {
+	if len(hand) > 6 || len(hand) < 4 {
+		return nil, errors.New(`hand size must be either 5 or 6`)
+	}
+
+	allHands, err := chooseNFrom(4, hand)
+	if err != nil {
+		return nil, err
+	}
+
+	summaries := []model.TossSummary{}
+
+	for _, h := range allHands {
+		tossed := without(hand, h)
+		handStats, cribStats := getStatsForHand(h, tossed)
+
+		summaries = append(summaries, model.TossSummary{
+			Kept:      h,
+			Tossed:    tossed,
+			HandStats: handStats,
+			CribStats: cribStats,
+		})
+	}
+
+	return summaries, nil
+}
+
+func getStatsForHand(
+	hand, tossed []model.Card,
+) (handStats, cribStats *tossStats) {
+	exclude := map[model.Card]struct{}{}
+	for _, c := range hand {
+		exclude[c] = struct{}{}
+	}
+	for _, c := range tossed {
+		exclude[c] = struct{}{}
+	}
+
+	handStats = &tossStats{}
+	cribStats = &tossStats{}
+	defer handStats.calculate()
+	defer cribStats.calculate()
+
+	for i := 0; i < 52; i++ {
+		lead := model.NewCardFromNumber(i)
+		if _, ok := exclude[lead]; ok {
+			continue
+		}
+
+		handStats.add(scorer.HandPoints(lead, hand))
+
+		exclude[lead] = struct{}{}
+		options := otherOptions(4-len(tossed), exclude)
```

Comment:
I'm struggling a bit to understand this myself, but let's see if I can explain it correctly: for each possible cut card (which is any card that _isn't_ in our hand), get the number of points we would have. Then, get all of the alternative options that we _could have_ tossed for this cut card and figure out what those scored would have been. Is that correct? What did I miss?

Reply:
No, wait, my bad. This bit is calculating all possible other cards that could be in the _crib_. Right?

Reply:
right - `options` is a slice of other possible entries into the crib